### PR TITLE
QtApplicationBase: Improve generalized usage of InviwoApplicationQt

### PR DESCRIPTION
**Pull request by ResearchDaniel**
Setting the icon in InviwoApplicationQt overrides icons set externally (e.g., using CMake).

This has been tested on: Mac, Windows

### CODE REVIEW

File: src/qt/applicationbase/inviwoapplicationqt.cpp [hunk 1]
```diff
@@ -135,19 +135,20 @@ void FileSystemObserverQt::fileChanged(QString fileName) {
...
+                                         std::string organizationName,
+                                         std::string organizationDomain)
     : QApplication(argc, argv)
     , InviwoApplication(argc, argv, displayName)
     , mainWindow_(nullptr)
     , uiLocal_(getCurrentStdLocale())
     , undoTrigger_{[]() {}} {
 
     setAttribute(Qt::AA_NativeWindows);
-    setWindowIcon(QIcon(":/inviwo/inviwo_light.png"));
```
Comment (martinfalk) [inline src/qt/applicationbase/inviwoapplicationqt.cpp:146]: does the default CMake file set the appropriate window icon?

Comment (ResearchDaniel) [reply src/qt/applicationbase/inviwoapplicationqt.cpp:146]: Yes, we set the following:

```
set(RES_FILES "")
if(WIN32)
    set(RES_FILES ${RES_FILES} "${IVW_RESOURCES_DIR}/inviwo.rc")
elseif(APPLE)
    set_source_files_properties(${IVW_ROOT_DIR}/resources/inviwo/inviwo_light.icns PROPERTIES MACOSX_PACKAGE_LOCATION Resources)
    set(RES_FILES ${RES_FILES} ${IVW_ROOT_DIR}/resources/inviwo/inviwo_light.icns)
endif()
```

I have not tested this on windows and Linux.

Comment (ResearchDaniel) [reply src/qt/applicationbase/inviwoapplicationqt.cpp:146]: Tested on Windows and there it seems like it is necessary to call setWindowIcon. So, I added an option to specify the icon

**Comment by ResearchDaniel** [comment]
Seems like there was an error when mergin. Will fix

**Pull request closed without merging**